diff --git a/arch/arm/mach-mv88de3100/modules/wlan_sd8787/Makefile b/arch/arm/mach-mv88de3100/modules/wlan_sd8787/Makefile
index ac77444..7928c9e 100644
--- a/arch/arm/mach-mv88de3100/modules/wlan_sd8787/Makefile
+++ b/arch/arm/mach-mv88de3100/modules/wlan_sd8787/Makefile
@@ -47,31 +47,31 @@ CONFIG_WIFI_DISPLAY_SUPPORT=y
 # Re-association in driver
 CONFIG_REASSOCIATION=y
 
-# Enable WEXT for STA
-CONFIG_STA_WEXT=y
-
-# Enable WEXT for uAP
-CONFIG_UAP_WEXT=y
-
-# Enable CFG80211 for STA
+# Pick CFG80211 or WEXT for STA
 ifeq ($(CONFIG_CFG80211),y)
 CONFIG_STA_CFG80211=y
+CONFIG_STA_WEXT=n
 else
 ifeq ($(CONFIG_CFG80211),m)
 CONFIG_STA_CFG80211=y
+CONFIG_STA_WEXT=n
 else
 CONFIG_STA_CFG80211=n
+CONFIG_STA_WEXT=y
 endif
 endif
 
-# Enable CFG80211 for uAP
+# Pick CFG80211 or WEXT for uAP
 ifeq ($(CONFIG_CFG80211),y)
 CONFIG_UAP_CFG80211=y
+CONFIG_UAP_WEXT=n
 else
 ifeq ($(CONFIG_CFG80211),m)
 CONFIG_UAP_CFG80211=y
+CONFIG_UAP_WEXT=n
 else
 CONFIG_UAP_CFG80211=n
+CONFIG_UAP_WEXT=y
 endif
 endif
 
