diff --git a/arch/arm/mach-mv88de3100/mv88de31xx_android.c b/arch/arm/mach-mv88de3100/mv88de31xx_android.c
index ddef649..a010988 100644
--- a/arch/arm/mach-mv88de3100/mv88de31xx_android.c
+++ b/arch/arm/mach-mv88de3100/mv88de31xx_android.c
@@ -70,16 +70,16 @@ static __initdata struct board_config {
 #define BG2PROTO_MTDPARTS(__RO__)                      \
       "mv_nand:"                                       \
       "1M(block0)ro," /* flash config params */        \
-      "8M(bootloader)" __RO__ "," /* 8x copies */      \
+      "8M(bootloader)ro," /* 8x copies */              \
       "7M(fts)ro,"                                     \
       "16M(kernel)"__RO__ ","                          \
       "60M(recovery),"                                 \
       "80M(backupsys)ro,"                              \
-      "16M(factory_store)" __RO__ ","                  \
+      "16M(factory_store)ro,"                          \
       "400M(rootfs)" __RO__ ","                        \
       "32M(localstorage),"                             \
       "300M(cache),"                                   \
-      "1024M(userdata),"                                \
+      "1024M(userdata),"                               \
       "8M@4088M(bbt)ro"
       /* TODO: limit write access to important partitions
        * in non-recovery mode
@@ -105,13 +105,13 @@ static __initdata struct board_config {
 #define EUREKA_B1_MTDPARTS(__RO__,__RRO__)             \
       "mv_nand:"                                       \
       "1M(block0)ro," /* flash config params */        \
-      "8M(bootloader)" __RO__ "," /* 8x copies */      \
+      "8M(bootloader)ro," /* 8x copies */              \
       "7M(fts)ro,"                                     \
       "16M(kernel)"__RO__ ","                          \
       "60M(recovery),"                                 \
       "80M(backupsys),"                                \
-      "16M(factory_store)" __RO__ ","                  \
-      "400M(rootfs)" __RRO__","                        \
+      "16M(factory_store)ro,"                          \
+      "400M(rootfs)" __RO__","                         \
       "300M(cache),"                                   \
       "1024M(userdata),"                               \
       "8M@2040M(bbt)ro"
@@ -136,7 +136,7 @@ static __initdata struct board_config {
 #define EUREKA_B2_MTDPARTS(__RO__)                           \
       "mv_nand:"                                             \
       "1M(block0)ro," /* flash config params */              \
-      "8M(bootloader)" __RO__ "," /* 8x copies */            \
+      "8M(bootloader)ro," /* 8x copies */                    \
       "16M(kernel)"__RO__ ","                                \
       "400M(rootfs)"__RO__ ","                               \
       "300M(cache),"                                         \
@@ -144,7 +144,7 @@ static __initdata struct board_config {
       "48M(recovery),"                                       \
       "96M(backupsys),"                                      \
       "8M(fts)ro,"                                           \
-      "16M(factory_store)" __RO__ ","                        \
+      "16M(factory_store)ro,"                                \
       "8M(bbt)ro"
       .mtdparts = EUREKA_B2_MTDPARTS("ro"),
       .mtdparts_ro = EUREKA_B2_MTDPARTS("ro"),
@@ -167,7 +167,7 @@ static __initdata struct board_config {
 #define EUREKA_B3_MTDPARTS(__RO__)                           \
       "mv_nand:"                                             \
       "1M(block0)ro," /* flash config params */              \
-      "8M(bootloader)" __RO__ "," /* 8x copies */            \
+      "8M(bootloader)ro," /* 8x copies */                    \
       "16M(kernel)"__RO__ ","                                \
       "400M(rootfs)"__RO__ ","                               \
       "300M(cache),"                                         \
@@ -175,7 +175,7 @@ static __initdata struct board_config {
       "48M(recovery),"                                       \
       "96M(backupsys),"                                      \
       "8M(fts)ro,"                                           \
-      "16M(factory_store)" __RO__ ","                        \
+      "16M(factory_store)ro,"                                \
       "8M(bbt)ro"
       .mtdparts = EUREKA_B3_MTDPARTS("ro"),
       .mtdparts_ro = EUREKA_B3_MTDPARTS("ro"),
