diff --git a/drivers/mtd/mtdblock.c b/drivers/mtd/mtdblock.c
index 4898481..4230971 100644
--- a/drivers/mtd/mtdblock.c
+++ b/drivers/mtd/mtdblock.c
@@ -340,7 +340,6 @@ static int mtdblock_open(struct mtd_blktrans_dev *mbd)
 	 */
 	mtdblk->block_map = NULL;
 	if (mbd->mtd->block_isbad &&
-		!(mbd->mtd->flags & MTD_WRITEABLE) &&
 		mbd->mtd->oobsize &&
 		mbd->mtd->erasesize &&
 		strcmp(mbd->mtd->name, ro_fspart_name) == 0) {
@@ -430,7 +429,7 @@ static void mtdblock_add_mtd(struct mtd_blktrans_ops *tr, struct mtd_info *mtd)
 	dev->mbd.size = mtd->size >> 9;
 	dev->mbd.tr = tr;
 
-	if (!(mtd->flags & MTD_WRITEABLE))
+	if (!(mtd->flags & MTD_WRITEABLE) || (strcmp(mtd->name, ro_fspart_name) == 0))
 		dev->mbd.readonly = 1;
 
 	if (add_mtd_blktrans_dev(&dev->mbd))
