diff --git a/arch/arm/mach-mv88de3100/mv88de31xx_android.c b/arch/arm/mach-mv88de3100/mv88de31xx_android.c
index 1f2f7ee..ddef649 100644
--- a/arch/arm/mach-mv88de3100/mv88de31xx_android.c
+++ b/arch/arm/mach-mv88de3100/mv88de31xx_android.c
@@ -189,7 +189,6 @@ static __initdata struct board_config {
 #define DEFAULT_BOARD  "bg2proto"
 static __initdata char board_name[32] = DEFAULT_BOARD;
 
-static __initdata char console[32];
 static __initdata char boot_mode[32];
 static __initdata char root[32];
 static __initdata int ro_mode = 0;
@@ -232,9 +231,6 @@ static int __init do_param(char *param, char *val)
         return 0;
     }
 
-    if (!strcmp(param, "console") && val)
-        strlcpy(console, val, sizeof(console));
-
     if (!strcmp(param, "root") && val)
         strlcpy(root, val, sizeof(root));
 
@@ -287,14 +283,19 @@ void __init mv88de31xx_android_fixup(struct tag *tag,
                         (unsigned int)tag);
         return;
     }
-    /* Override from(defaultp_command_line) */
-    *from = mv88de31xx_command_line;
 
     /* Scan boot command line removing all memory layout
      * parameters on the way.
      */
     mv88de31xx_command_line[0] = '\0';
+
+#if defined(CONFIG_CMDLINE_EXTEND) || defined(CONFIG_CMDLINE_FORCE)
+    strlcat(mv88de31xx_command_line, *from, COMMAND_LINE_SIZE);
+#endif
+
+#if defined(CONFIG_CMDLINE_EXTEND) || defined(CONFIG_CMDLINE_FROM_BOOTLOADER)
     parse_args("memory_setup", cmdline, NULL, 0, do_param);
+#endif
 
     recovery_boot = !strcmp(boot_mode, "recovery");
     if (recovery_boot)
@@ -325,13 +326,6 @@ void __init mv88de31xx_android_fixup(struct tag *tag,
     add_boot_param("androidboot.hardware", board_name);
     if (boot_mode[0])
         add_boot_param("androidboot.mode", boot_mode);
-    if (console[0]) {
-        /* keep only console device name, e.g. ttyS0 */
-        char *p = strchr(console, ',');
-        if (p)
-            *p = '\0';
-        add_boot_param("androidboot.console", console);
-    }
 
     /* Register optional reboot hook */
     if (cfg->reboot_notifier) {
@@ -339,6 +333,9 @@ void __init mv88de31xx_android_fixup(struct tag *tag,
         register_reboot_notifier(&reboot_notifier);
     }
 
+    /* Override from(defaultp_command_line) */
+    *from = mv88de31xx_command_line;
+
     return;
 }
 
